src/elements/elements.hpp

```diff
@@ -0,0 +1,110 @@
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+#include "src/elements/beams/beams.hpp"
+#include "src/elements/masses/masses.hpp"
+
+namespace openturbine {
+
+/**
+ * @brief A container providing handle to all structural elements present in the model
+ *
+ * @details This class serves as a container for wrapping all structural elements
+ *          in the model, including beams and point masses, effectively providing
+ *          the entire mesh as a single object to be used by the gen-alpha solver
+ *          logic.
+ *
+ * @note The class ensures that at least one type of element (beams or masses) is present
+ *       in the model
+ */
+struct Elements {
+    std::shared_ptr<Beams> beams;
```

Comment:
I agree that Elements should own its members - semantically, that should be the correct design.  They shouldn't be owned anywhere else in the code, so something like a shared_ptr isn't needed.

Reply:
> What's the reasoning for these being shared pointers? Are they going to be referenced from somewhere besides the Element object?

> I agree that Elements should own its members - semantically, that should be the correct design. They shouldn't be owned anywhere else in the code, so something like a shared_ptr isn't needed.

Currently, our elements (i.e. `Beams`) are not owned by the `Model` class, but the `Nodes` are. Going forward, I was envisioning a design where `Model` class creates and owns all the resources including `Elements`. From that perspective, it made sense to design `Elements` as sharing handles to the underlying resources (could be references/pointers). Having said that, I did not actually make any changes to the `Model` class design here (I think it's a separate issue to make our model creation consistent). If you think we should rather design `Elements` to own its members, I believe it will be a rather quick pivot to make -- please respond here to let me know.

Reply:
I still think `Elements` should own its members.  If `Model` owns `Elements` or it is stand alone is another discussion we need to have, but either way the members just need to live inside the single struct.

Reply:
Let me think about how to simplify this logic -- I'll prioritize getting this functional first.